target/linux/ath79/image/generic-tp-link.mk

```diff
@@ -275,6 +275,52 @@ define Device/tplink_archer-d50-v1
 endef
 TARGET_DEVICES += tplink_archer-d50-v1
 
+define Device/tplink_archer-d7-v1
```

Comment:
I think in this case it's worth creating a tplink_archer-d7-common ...

Reply:
What do you think about this? https://github.com/DavideFioravanti/openwrt/commit/c436654c669c0bd25043d51caafba74c79b8c8c7
We could use this common device also for the Archer D50

Reply:
Generally yes, but is it really "v3" when I see lines like that:
`tplink-v2-header -s -V "ver. 2.0"`
@ynezz @blocktrron Can you provide a quick opinion?

Reply:
I would go for something like `tplink-qca-dsl`. It is definitely not a v3 header.

Also, the `-V "ver. 2.0"` option seems redundant to me: https://github.com/openwrt/openwrt/blob/master/include/image-commands.mk#L337

Reply:
As far as I know these devices are indeed based on the v3 header. According to [this page](https://github.com/xdarklight/mktplinkfw3):

> The value of the version field is 0x03 0x00 0x00 0x00 and the value of the fw_version field starts with ver. 2.0.

However the "tplink-qca-dsl" should fit every case ;)